src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -0,0 +1,142 @@
+import no.nav.familie.ef.sak.oppgave.OppgaveService
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.PersonopplysningerIntegrasjonerClient
+import no.nav.familie.kontrakter.felles.Behandlingstema
+import no.nav.familie.kontrakter.felles.Fødselsnummer
+import no.nav.familie.kontrakter.felles.Tema
+import no.nav.familie.kontrakter.felles.ef.StønadType
+import no.nav.familie.kontrakter.felles.oppgave.IdentGruppe
+import no.nav.familie.kontrakter.felles.oppgave.OppgaveIdentV2
+import no.nav.familie.kontrakter.felles.oppgave.Oppgavetype
+import no.nav.familie.kontrakter.felles.oppgave.OpprettOppgaveRequest
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Locale
+
+@Service
+class BarnFyllerÅrOppfølgingsoppgaveService(
+    private val gjeldendeBarnRepository: GjeldendeBarnRepository,
+    private val oppgaveClient: OppgaveClient,
+    private val oppgaveService: OppgaveService,
+    private val oppgaveRepository: OppgaveRepository,
+    private val personopplysningerIntegrasjonerClient: PersonopplysningerIntegrasjonerClient
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun opprettOppgaverForAlleBarnSomHarFyltÅr(dryRun: Boolean = false) {
+        val dagensDato = LocalDate.now()
+        val alleBarnIGjeldendeBehandlinger =
+            gjeldendeBarnRepository.finnBarnAvGjeldendeIverksatteBehandlinger(StønadType.OVERGANGSSTØNAD, dagensDato) +
+                gjeldendeBarnRepository.finnBarnTilMigrerteBehandlinger(StønadType.OVERGANGSSTØNAD, dagensDato)
+
+        logger.info("Antall barn i gjeldende behandlinger: ${alleBarnIGjeldendeBehandlinger.size}")
+
+        val skalOpprettes = filtrerBarnSomHarFyltÅr(alleBarnIGjeldendeBehandlinger)
+
+        if (!dryRun) {
+            opprettOppgaveForBarn(skalOpprettes)
+        } else {
+            logger.info("Ville opprettet oppgave for ${skalOpprettes.size} barn.")
+            skalOpprettes.forEach {
+                secureLogger.info(
+                    "Ville opprettet oppgave for barn med fødselsnummer: " +
+                        "${it.fødselsnummer} med alder ${it.alder}"
+                )
+            }
+        }
+    }
+
+    private fun filtrerBarnSomHarFyltÅr(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): List<OpprettOppgaveForBarn> {
+        val opprettedeOppgaver = oppgaveRepository.findByTypeAndAlderIsNotNull(Oppgavetype.InnhentDokumentasjon)
+        val skalOpprettes = mutableListOf<OpprettOppgaveForBarn>()
+
+        barnTilUtplukkForOppgave.forEach { barn ->
+            val barnetsAlder = Alder.fromFødselsdato(fødselsdato(barn))
+            if (barnetsAlder != null && barn.fødselsnummerBarn != null && opprettedeOppgaver.none { it.barnPersonIdent == barn.fødselsnummerBarn && it.alder == barnetsAlder }) {
```

Comment:
Hvorfor trenger vi å sjekke `it.alder == barnetAalder`

Reply:
For at det ikke skal opprettes oppgave på et barn som allerede har fylt år og som det har blitt opprettet oppgave på.
Første utplukk fra db tar ikke hensyn til hvilke barn det er opprettet oppgave for (allerede kompleks nok spørring fra før, så har unngått å utvide den med vilje for lesbarhetens og eventuell feilsøking sin del)